src/main/java/com/salaboy/conferences/agenda/controller/AgendaController.java

```diff
@@ -0,0 +1,86 @@
+import io.zeebe.client.api.response.ActivatedJob;
+import io.zeebe.client.api.worker.JobClient;
+import io.zeebe.spring.client.annotation.ZeebeWorker;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.*;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RestController
+@Slf4j
+@RequestMapping
+public class AgendaController {
+
+    private final AgendaItemRepository agendaItemRepository;
+    private final ObjectMapper objectMapper;
+
+    public AgendaController(AgendaItemRepository agendaItemRepository, ObjectMapper objectMapper) {
+        this.agendaItemRepository = agendaItemRepository;
+        this.objectMapper = objectMapper;
+    }
+
+    @PostMapping
+    public Mono<String> newAgendaItem(@RequestBody AgendaItem agendaItem) {
+
+        log.info("> New Agenda Item Received: " + agendaItem);
+
+        if (agendaItem.getTitle().contains("fail")) {
+            throw new IllegalStateException("Something went wrong with adding the Agenda Item: " + agendaItem);
+        }
+
+        Mono<String> result = agendaItemRepository.save(agendaItem)
+                .doOnSuccess(i -> log.info("> Agenda Item Added to Agenda: {}", i))
+                .doOnError(i -> log.info("> Agenda Item NOT Added to Agenda: {}", i))
+                .map(i -> !Strings.isNullOrEmpty(i.getId()) ? "Agenda Item Added to Agenda" : "Agenda Item Not Added");
+
+        result.subscribe();
+
+        return result;
+    }
+
+    @GetMapping
+    public Flux<AgendaItem> getAll() {
+        return agendaItemRepository.findAll();
+    }
+
+    @GetMapping("/day/{day}")
+    public Mono<Set<AgendaItem>> getAllByDay(@PathVariable(value = "day", required = true) final String day) {
+        return agendaItemRepository.findAllByDay(day).collect(Collectors.toSet());
+    }
+
+    @GetMapping("/{id}")
+    public Mono<AgendaItem> getById(@PathVariable("id") String id) {
+
+        return agendaItemRepository.findById(id);
+    }
+
+    @DeleteMapping("/")
+    public Mono<Void> clearAgendaItems() {
+
+        log.info(">>> Deleting all");
+        return agendaItemRepository.deleteAll();
+    }
+
+    @ZeebeWorker(name = "agenda-worker", type = "agenda-publish")
+    public void newAgendaItemJob(final JobClient client, final ActivatedJob job) {
+        Proposal proposal = objectMapper.convertValue(job.getVariablesAsMap().get("proposal"), Proposal.class);
+        String[] days = {"Monday", "Tuesday"};
+        String[] times = {"9:00 am", "10:00 am", "11:00 am", "1:00 pm", "2:00 pm", "3:00 pm", "4:00 pm", "5:00 pm"};
+        Random random = new Random();
+        int day = random.nextInt(2);
+        int time = random.nextInt(8);
+        newAgendaItem(new AgendaItem(proposal.getTitle(), proposal.getAuthor(), days[day], times[time]));
+        client.newCompleteCommand(job.getKey()).send();
```

Comment:
@mcruzdev if we want to test this line here.. we need to mock it in the same way that we are mocking the ZeebeLifecycleClient in the c4p service right?